pkg/utils/grpc.go

```diff
@@ -42,9 +49,29 @@ func CreateGrpcConnection(ctx context.Context, c client.Client, pod *v1.Pod) (*g
 		return nil, err
 	}
 
-	conn, err := grpc.Dial(fmt.Sprintf("%s:%s", node.Status.Addresses[0].Address, port), grpc.WithInsecure())
+	conn, err := grpc.Dial(fmt.Sprintf("%s:%s", node.Status.Addresses[0].Address, port),
+		grpc.WithInsecure(),
+		grpc.WithUnaryInterceptor(TimeoutClientInterceptor))
 	if err != nil {
 		return nil, err
 	}
 	return conn, nil
 }
+
+// TimeoutClientInterceptor wraps the RPC with a timeout.
+func TimeoutClientInterceptor(ctx context.Context, method string, req, reply interface{},
+	cc *grpc.ClientConn, invoker grpc.UnaryInvoker, opts ...grpc.CallOption) error {
+	ctx, cancel := context.WithTimeout(ctx, time.Duration(RPCTimeout)*time.Millisecond)
+	defer cancel()
+	return invoker(ctx, method, req, reply, cc, opts...)
+}
+
+// TimeoutServerInterceptor ensures the context is intact before handling over the
+// request to application.
+func TimeoutServerInterceptor(ctx context.Context, req interface{}, info *grpc.UnaryServerInfo,
```

Comment:
What's the purpose of TimeoutServerInterceptor here? seems that's not necessary?

Reply:
In case that a lately arrived RPC because of unstable network condition, it will abandon the RPC when it's already timeout.

Reply:
Got it, thanks.